lib/gateway.js

```diff
@@ -146,6 +144,25 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
   }
 }
 
+function makeEntityTransformData (field) {
+  const isNonNull = field.astNode.type.kind === Kind.NON_NULL_TYPE
+  const leafKind = isNonNull ? field.astNode.type.type.kind : field.astNode.type.kind
+
+  if (leafKind === Kind.LIST_TYPE) {
+    return function transformData (response) {
+      if (!response) {
+        return isNonNull ? [] : null
+      }
+
+      return response.json.data._entities
+    }
+  }
+
+  return function transformData (response) {
+    return response.json.data._entities[0]
+  }
```

Comment:
This function can be a top level one, as there is no need to allocate a closure.